### PR TITLE
RavenDB-20028 Databases view - filter by state

### PR DESCRIPTION
**Pull request by kalczur**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-20028/Databases-view-design

### Additional description

- added logic to MultiCheckboxToggle and MultiRadioToggle
- added DatabasesFilter component
- added stories for Toggles
- removed DatabaseCount component

For database filter by state we have 3 groups:
- state
- sharded/nonsharded
- local/remote

for checking a condition within a group is logical OR, and between groups AND

### Type of change

- New feature

### How risky is the change?

- Low

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing by Contributor

- It has been verified by manual testing

### Testing by RavenDB QA team

- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- No

### UI work

image

### CODE REVIEW

File: src/Raven.Studio/typescript/components/common/MultiCheckboxToggle.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    return (
+        <div className={classNames("multi-toggle", className)}>
+            {label && <div className="small-label ms-1 mb-1">{label}</div>}
+            <div className="multi-toggle-list">
+                {itemSelectAll && (
+                    <div className="multi-toggle-item">
+                        <input
+                            id={uniqueId + itemSelectAll.label}
+                            type="checkbox"
+                            name={uniqueId + itemSelectAll.label}
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/MultiCheckboxToggle.tsx]: should we use label or value in field name?

File: src/Raven.Studio/typescript/components/common/MultiCheckboxToggle.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    };
+
+    return (
+        <div className={classNames("multi-toggle", className)}>
+            {label && <div className="small-label ms-1 mb-1">{label}</div>}
+            <div className="multi-toggle-list">
+                {itemSelectAll && (
+                    <div className="multi-toggle-item">
+                        <input
+                            id={uniqueId + itemSelectAll.label}
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/MultiCheckboxToggle.tsx]: same here

File: src/Raven.Studio/typescript/components/common/MultiCheckboxToggle.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                            type="checkbox"
+                            name={uniqueId + itemSelectAll.label}
+                            checked={isSelectedAll}
+                            onChange={onChangeSelectAll}
+                        />
+                        <label htmlFor={uniqueId + itemSelectAll.label}>
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/MultiCheckboxToggle.tsx]: and here

File: src/Raven.Studio/typescript/components/common/MultiCheckboxToggle.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+                                {itemSelectAll.label}
+                                {itemSelectAll.count >= 0 && (
+                                    <span className="multi-toggle-item-count">{itemSelectAll.count}</span>
+                                )}
+                            </span>
+                        </label>
+                    </div>
+                )}
+                {inputItems.map((inputItem) => (
+                    <div className="multi-toggle-item" key={uniqueId + inputItem.label}>
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/MultiCheckboxToggle.tsx]: and here

Comment (ml054) [reply src/Raven.Studio/typescript/components/common/MultiCheckboxToggle.tsx]: and few places below

File: src/Raven.Studio/typescript/components/common/RadioToggle.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import classNames from "classnames";
+
+interface ToggleProps {
+    className?: string;
+}
+
+// TODO: add whole logic
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/RadioToggle.tsx:8]: Is it reported?

File: src/Raven.Studio/typescript/components/common/Toggles.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        { value: "1day", label: "1 day" },
+    ];
+
+    const checkboxList: InputItem[] = [
+        { value: "normal", label: "Normal" },
+        { value: "error", label: "Error/Faulty", count: 3 },
+        { value: "stale", label: "Stale" },
+        { value: "rolling", label: "Rolling deployment", count: 1 },
+        { value: "idle", label: "Idle", count: 0 },
+        { value: "local", label: "Local", verticalSeparatorLine: true },
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/Toggles.stories.tsx:31]: nice API :)

File: src/Raven.Studio/typescript/components/common/shell/databasesSlice.ts [hunk 7]
```diff
@@ -119,27 +123,82 @@ export const selectDatabasesSummary = (store: RootState) => {
...
+
+const isDatabaseInFilterState = (
+    store: RootState,
+    db: DatabaseSharedInfo,
+    filterStates: DatabaseFilterByStateOption[]
+): boolean => {
+    const perNodeState = selectDatabaseState(db.name)(store);
+    const databaseState = DatabaseUtils.getDatabaseState(db, perNodeState);
+
+    if (
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/shell/databasesSlice.ts]: no need for if/return true/return false

just return. 

Optionally we might split that into separate conditions for readability

File: src/Raven.Studio/typescript/components/common/shell/databasesSlice.ts [hunk 8]
```diff
@@ -119,27 +123,82 @@ export const selectDatabasesSummary = (store: RootState) => {
...
     const criteria = selectDatabaseSearchCriteria(store);
     const databases = selectAllDatabases(store);
 
-    if (criteria.searchText) {
-        return databases.filter((x) => x.name.toLowerCase().includes(criteria.searchText.toLowerCase()));
+    if (!(criteria.name || criteria.states?.length > 0)) {
+        return databases;
+    }
+
+    let filteredDatabases = [...databases];
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/shell/databasesSlice.ts]: no need for shallow copy here?

File: src/Raven.Studio/typescript/components/common/shell/databasesSlice.ts [hunk 9]
```diff
@@ -277,8 +336,14 @@ export const databasesSlice = createSlice({
             //TODO: update in shallow mode?
             databasesAdapter.setAll(state.databases, action.payload);
         },
-        filterTextSet: (state, action: PayloadAction<string>) => {
-            state.searchCriteria.searchText = action.payload;
+        localNodeTagChanged: (state, action: PayloadAction<string>) => {
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/shell/databasesSlice.ts:343]: local node tag can't ever be changed?

File: src/Raven.Studio/typescript/components/common/RadioToggle.tsx (hunk 5 above)
Comment (kalczur) [reply src/Raven.Studio/typescript/components/common/RadioToggle.tsx:8]: https://issues.hibernatingrhinos.com/issue/RavenDB-20254/Add-logic-to-the-RadioToggle-react-component

File: src/Raven.Studio/typescript/components/common/shell/setup.ts [hunk 10]
```diff
@@ -21,4 +22,5 @@ export function initRedux() {
 
     databasesManager.default.onUpdateCallback = throttledUpdateReduxStore;
     activeDatabaseTracker.default.database.subscribe((db) => globalDispatch(activeDatabaseChanged(db?.name ?? null)));
+    clusterTopologyManager.default.localNodeTag.subscribe((nodeTag) => globalDispatch(localNodeTagChanged(nodeTag)));
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/shell/setup.ts:27]: ok let's leave it like that but put note to fetch that from redux (we should create seprate slice for cluster info).

File: src/Raven.Studio/typescript/components/common/MultiCheckboxToggle.tsx (hunk 1 above)
Comment (kalczur) [reply src/Raven.Studio/typescript/components/common/MultiCheckboxToggle.tsx]: value does not necessarily have to be a string, but label does

File: src/Raven.Studio/typescript/components/common/shell/databasesSlice.ts (hunk 8 above)
Comment (kalczur) [reply src/Raven.Studio/typescript/components/common/shell/databasesSlice.ts]: yes, you're right

File: src/Raven.Studio/typescript/components/common/shell/databasesSlice.ts (hunk 7 above)
Comment (kalczur) [reply src/Raven.Studio/typescript/components/common/shell/databasesSlice.ts]: maybe I will improve readability with formating and use // prettier-ignore

Comment (ml054) [reply src/Raven.Studio/typescript/components/common/shell/databasesSlice.ts]: ok let's leave it

File: src/Raven.Studio/typescript/components/common/MultiCheckboxToggle.tsx (hunk 1 above)
Comment (ml054) [reply src/Raven.Studio/typescript/components/common/MultiCheckboxToggle.tsx]: ok

**Comment by ml054** [comment]
@kalczur please use values instead of labels.

**Comment by kalczur** [comment]
Update: now displays loading databases regardless of filtered state

**Review by ml054 — APPROVED** [review]

**Pull request merged**